plugin/openassetio_manager_bal/BasicAssetLibraryInterface.py

```diff
@@ -43,6 +45,20 @@
 # pylint: disable=too-many-arguments, unused-argument
 
 
+def simulated_delay(func):
+    @wraps(func)
+    def wrapper_simulated_delay(self, *args, **kwargs):
+        # pylint: disable=protected-access
+        delay_ms = self._BasicAssetLibraryInterface__settings.get(
+            bal.SETTINGS_KEY_SIMULATED_QUERY_LATENCY, 0
```

Comment:
[consider] There doesn't seem to be a test for the case that the setting is not set at all?

Perhaps a little test suite for the decorator itself would be good.

Reply:
There isn't really a way this setting can be unset, the `.get()` here is an exercise in paranoid programming. 

(I'm always torn on this, would be interested in hearing opinions. On the one hand, I think defensive programming is better for the released artifact, it's more resilient under change from the users perspective, but it does have a detriment re: code-as-documentation of intent.)

If you attempt to overwrite the default with anything other than an `int` (and now a `float`), the `validate_settings` call will fail. Due to that I don't think we need to test that case.

Reply:
That's a nice additional check :+1: 

But my original question was more about if the setting wasn't set at all. Unless I'm being blind, I don't see a unit test where `"simulated_query_latency_ms"` is not set, i.e. testing that we fall back to zero.

Reply:
Still not sure we're quite on the same page, as `simulated_query_latency_ms` will never fall back to zero, the default always gets set (defined as 10ms,) and there's no publicly accessible way to avoid this.
Did you mean testing that? Added : https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/39/commits/807786c12b15edfc8371df826bfab3c0982f1d15

(If there's another path you're seeing we should get on a call because i'm totally missing it.)

Reply:
This discussion also gave me the thought to test the partial application behaviour more explicitly too, so I added : https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/39/commits/a21f63dd26a0d1e81149d2fba94644bcea3b6215